frontend/src/pages/JobFeed/JobFeed.tsx

```diff
@@ -55,10 +54,10 @@ export default function JobFeed() {
   function handleChange() {
     if (selectedSort === sortStatus.ASC) {
       setselectedSort(sortStatus.DEC);
-      data?.data?.sort((a, b) => a.createdDate.localeCompare(b.createdDate));
+      data?.data?.sort((a, b) => a.createdDate.toLocaleString().localeCompare(b.createdDate.toLocaleString()));
```

Comment:
.... And now I finally get to why all that typing nonsense actually matters. Because here is a bug, it was a bug before and now it's an even more subtle bug. First, I want to make explicit what's happening here. Date#toLocaleString turns dates to strings based on the browser's locale (language/location) settings. string#localeCompare then compares those strings *alphabetically* based on the browser's locale.

In my locale, Date#toLocaleString gives: "11/10/2021, 9:34:15 PM"*. So, "11/09/2022, 9:34:15 PM" (ie exactly 364 days in the future) is alphabetically *prior* to the first date. This may not seem so bad, but what if my locale printed dates like "Jan 1"? Then "Feb 1" comes before "Jan 1".

Proposed solution: Just use `<`. It is overloaded for JS Date. Remember, this requires that we *do in fact parse the dates* from the response. Otherwise it will compare strings and we're back where we started (although it may happen to work because of the default date format. Still worth doing it right).

\* (Yes, go ahead and compare this to when this review was submitted, and this came from new Date() a while into writing this. I've been working on this review long enough. Yeah, I put in the work even when I'm busy during class.)

Reply:
I am leaning toward using the built-in Date parser. For the interface, there should be two different types for the jobPost: JobPostResponse(json) and JobPost(actual) I think.

Reply:
we can use Date.toJSON() instead then?

Reply:
Sunwoo: i agree, that's what i proposed in the above comment
Dan: i'm not sure i understand. We *do* use Date.toJSON (on the server) (we don't explicitly use it, we use response.json(...), which calls JSON.stringify, which calls Date.toJSON)

Reply:
The point is that when the frontend needs to use a non-string field (such as Date field), does the frontend want to use it as its original type or as string type. If it wants to use the original type, then we need a parser that changes the type from string to the original type.

Reply:
MM, yeah, you could leave it as a string, but that's basically only reasonable for this particular task. Even for displaying the dates we need to convert it to a JS Date (unless you want to display it in ISO format haha)

Reply:
Update 5 days later... this bug is now obsolete since Jake will be moving the sorting to the backend and Prisma will handle the sorting. That being said, we still need to choose between date and string..